### PR TITLE
Active contracts

### PR DESCRIPTION
**Pull request by sedaghatfar**
# Description

_Please include a summary of changes and related issue (if any)._
A table tracking the number of active contracts deployed

# Tests

- [ ] Please provide evidence of your successful `dbt run` / `dbt test` here
image
image

- [ ] Any comparison between `prod` and `dev` for any schema change

# Checklist
- [ ] Follow [dbt style guide](https://github.com/dbt-labs/corp/blob/main/dbt_style_guide.md)
- [ ] Tag the person(s) responsible for reviewing proposed changes
@0xissui 
- [ ] Notes to deployment, if a `full-refresh` is needed for any table
- [ ] Run `git merge main` to pull any changes from remote into your branch prior to merge.

### CODE REVIEW

**Comment by forgxyz** [comment]
The transaction receiver is not necessarily a contract. In a P2P NEAR transfer, this method is counting individuals as contracts.

**Comment by sedaghatfar** [comment]
> The transaction receiver is not necessarily a contract. In a P2P NEAR transfer, this method is counting individuals as contracts.

Correct, this is based off https://github.com/near/near-analytics/blob/main/aggregations/db_tables/daily_active_contracts_count.py in individuals using P2P are  contracts as well

**Comment by sedaghatfar** [comment]
Near has a different naming system than other blockchains see here [Exploring NEAR Protocol APIs](https://github.com/near-examples/workshop--exploring-near-apis#accounts)

Example of a P2P transfer not being counted as a contract 
tx_tash - 6ahKTLaG7ga5Zh3ayUMQVmj43EnDdjauoNS23wBMuD3Z

**Comment by sedaghatfar** [comment]
Near has a different naming system than other blockchains see here [Exploring NEAR Protocol APIs](https://github.com/near-examples/workshop--exploring-near-apis#accounts)

Example of a P2P transfer not being counted as a contract 
tx_tash - 6ahKTLaG7ga5Zh3ayUMQVmj43EnDdjauoNS23wBMuD3Z

**Comment by sedaghatfar** [comment]
Near has a different naming system than other blockchains see here [Exploring NEAR Protocol APIs](https://github.com/near-examples/workshop--exploring-near-apis#accounts)

Example of a P2P transfer not being counted as a contract 
tx_hash - 6ahKTLaG7ga5Zh3ayUMQVmj43EnDdjauoNS23wBMuD3Z
not showing any data in the NEAR_DEV.SILVER.ACTIONS_EVENTS_FUNCTION_CALL table
same for tx_hash FD5yF6X4tbUjYtfZkPe9ZwDQby5bkQ4sTbjMrryhaMw5 transfer from sweat welcome to another address
tx_hash - Die5nz5ZLaWat52QTyUQWoG5E5TBNTCqvCsJCFfEqxqP
P2P both accounts have a name
Example that is being counted as an active contact 3qKBCCg3He2o4QufUVYVLgUrjEoaZz5fK2suqckrgzA4
using the near crowd contact

**Comment by sedaghatfar** [comment]
What we might want to do is exclude method names like transfer, but could cause later metics to not match
```SELECT 
    method_name
    , COUNT(*) as transactions
    , COUNT(distinct t.tx_receiver) as distinct_receivers
FROM NEAR_DEV.SILVER.TRANSACTIONS t 
JOIN NEAR_DEV.SILVER.ACTIONS_EVENTS_FUNCTION_CALL aefc ON aefc.tx_hash = t.tx_hash
WHERE t.block_timestamp > CURRENT_DATE - 90
GROUP BY 1
ORDER BY 3 DESC
```
image

File: models/metrics/metrics__active_contracts.sql [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+  FROM
+    {{ ref('silver__transactions') }}
+    t
+    JOIN {{ ref('silver__actions_events_function_call') }}
+    aefc
+    ON t.tx_hash = aefc.tx_hash
+  WHERE
+    1 = 1
+
+{% if is_incremental() %}
```
Comment (0xissui) [inline models/metrics/metrics__active_contracts.sql:22]: do we need this if we already set as incremental in line 2?

File: models/metrics/metrics__active_contracts.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
+version: 2
+
+models:
+  - name: metrics__active_contracts
+    description: "Daily, weekly, and monthly active contracts on NEAR."
+
+    columns:
+      - name: DATE
+        description: The date (day).
```
Comment (0xissui) [inline models/metrics/metrics__active_contracts.yml:9]: Let's use the markdown tables in the descriptions folder (make one if doesn't exist yet!) 
e..g ` "{{ doc('date')}}" `

**Review by 0xissui — CHANGES_REQUESTED** [review]
overall good after nits!

File: models/metrics/metrics__active_contracts.sql (hunk 1 above)
Comment (sedaghatfar) [reply models/metrics/metrics__active_contracts.sql:22]: protective if you do a full refresh

File: models/metrics/metrics__active_contracts.yml (hunk 2 above)
Comment (sedaghatfar) [reply models/metrics/metrics__active_contracts.yml]: Added, thank you

**Review by 0xissui — APPROVED** [review]
LGTM

**Pull request merged**